lib/datadog/statsd/uds_connection.rb

```diff
@@ -14,18 +14,26 @@ def initialize(socket_path, **kwargs)
         super(**kwargs)
 
         @socket_path = socket_path
+        @socket = nil
+        connect
+      end
+
+      def close
+        @socket.close if @socket
+        @socket = nil
       end
 
       private
 
       def connect
-        socket = Socket.new(Socket::AF_UNIX, Socket::SOCK_DGRAM)
-        socket.connect(Socket.pack_sockaddr_un(@socket_path))
-        socket
+        close unless @socket == nil
+
```

Comment:
Minor: For consistency with the udp connection class:

```suggestion
        close if @socket

```